renderlib/io/FileReaderCCP4.cpp

```diff
@@ -165,16 +169,78 @@ readCCP4Dimensions(const std::string filepath, VolumeDimensions& dims, uint32_t
     case 6:
       dims.bitsPerPixel = 16;
       dims.sampleFormat = 1;
+    case 12:
+      dims.bitsPerPixel = 16;
+      dims.sampleFormat = 3;
     default:
       LOG_ERROR << "Bad file read from " << filepath;
       return false;
   }
 
-  dims.sizeC = 1;
-  dims.sizeT = 1;
+  uint32_t dummy;
+  // NCSTART         Number of first COLUMN  in map
+  myFile.read((char*)&dummy, 4);
+  if (!myFile || myFile.gcount() != 4) {
+    LOG_ERROR << "Bad file read from " << filepath;
+    return false;
+  }
+  // NRSTART         Number of first ROW     in map
+  myFile.read((char*)&dummy, 4);
+  if (!myFile || myFile.gcount() != 4) {
+    LOG_ERROR << "Bad file read from " << filepath;
+    return false;
+  }
+  // NSSTART         Number of first SECTION in map
+  myFile.read((char*)&dummy, 4);
+  if (!myFile || myFile.gcount() != 4) {
+    LOG_ERROR << "Bad file read from " << filepath;
+    return false;
+  }
+  // NX              Number of intervals along X
+  myFile.read((char*)&dummy, 4);
+  if (!myFile || myFile.gcount() != 4) {
+    LOG_ERROR << "Bad file read from " << filepath;
+    return false;
+  }
+  // NY              Number of intervals along Y
+  myFile.read((char*)&dummy, 4);
+  if (!myFile || myFile.gcount() != 4) {
+    LOG_ERROR << "Bad file read from " << filepath;
+    return false;
+  }
+  // NZ              Number of intervals along Z
+  myFile.read((char*)&dummy, 4);
+  if (!myFile || myFile.gcount() != 4) {
+    LOG_ERROR << "Bad file read from " << filepath;
+    return false;
+  }
```

Comment:
A lot of repeated code here it looks like... can this be a for loop?

Reply:
Yeah I could probably move this into a function or sometihng.  Or maybe just keep track of the data instead of using dummy vars.

Reply:
Opting to leave this as-is for later when I will very likely read the values into actual variables to be used.